File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.cpp
class Solution {
public:
    int firstStableIndex(vector<int>& nums, int k) {
        int mx = INT_MIN;
        int index = -1;

        for(int i=0; i<nums.size(); i++){
            mx = max(mx, nums[i]);

            int mn = INT_MAX;
            for(int j=i; j<nums.size(); j++){
                mn = min(mn, nums[j]);
            }

            int score = (mx-mn);
            if(score <= k){
                index = i;
                break;
            }
        }

        return index;
    }
};

File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.java
class Solution {
    public int firstStableIndex(int[] nums, int k) {
        int max = Integer.MIN_VALUE;
        int index = -1;

        for(int i=0; i<nums.length; i++){
            max = Math.max(max, nums[i]);

            int min = Integer.MAX_VALUE;
            for(int j=i; j<nums.length; j++){
                min = Math.min(min, nums[j]);
            }

            int score = (max-min);
            if(score <= k){
                index = i;
                break;
            }
        }

        return index;
    }
}